Add how-to for configuring XFF handling (#279)
appuio/openshift4-docs: docs/modules/ROOT/pages/how-tos/ingress/xff-configuration.adoc

= Configuring X-Forwarded-For handling

> This how-to explains how to configure the ingress' handling of the forwarded headers (`Forwarded` and `X-Forwarded-For`) on a per-route basis. 

Reviewer: ```suggestion
This how-to explains how to customize how the forwarding headers (`Forwarded` and `X-Forwarded-For`) are handled by the OpenShift ingress router on a per-route basis. 
```

== Possible behaviours

> * *`append`*: By default, the ingress is configured to append the headers to any existing headers. If `Forwarded` and `X-Forwarded-For` headers are already present when the ingress receives a request, the existing values are preserved, and new values are appended by the ingress.

Reviewer: ```suggestion
* *`append`*: By default, the OpenShift router is configured to append the headers to any existing headers.
If `Forwarded` and `X-Forwarded-For` headers are already present when the OpenShift router receives a request, the existing values are preserved, and new values are appended by the router.
```

> * *`append`*: By default, the ingress is configured to append the headers to any existing headers. If `Forwarded` and `X-Forwarded-For` headers are already present when the ingress receives a request, the existing values are preserved, and new values are appended by the ingress.

Reviewer: Consider using the following formatting:

```suggestion
`append`:: By default, the ingress is configured to append the headers to any existing headers. If `Forwarded` and `X-Forwarded-For` headers are already present when the ingress receives a request, the existing values are preserved, and new values are appended by the ingress.
```

> * *`replace`*: If forwarded header handling is set to `replace`, any existing forwarded headers are discarded. Only the headers set by the ingress will be available.

Reviewer: ```suggestion
* *`replace`*: If forwarded header handling is set to `replace`, any existing forwarded headers are discarded.
Only the headers set by the OpenShift router will be available.
```

> * *`never`*: If forwarded header handling is set to `never`, the ingress will never set these headers, and leave any existing headers untouched.

Reviewer: ```suggestion
* *`never`*: If forwarded header handling is set to `never`, the OpenShift router will never set these headers, and leave any existing headers untouched.
```

> * *`if-none`*: If forwarded header handling is set to `if-none`, the ingress will only set the headers if they're not already present. If any headers are present, they're left unchanged.

Reviewer: ```suggestion
* *`if-none`*: If forwarded header handling is set to `if-none`, the OpenShift router will only set the headers if they're not already present.
If any headers are present, they're left unchanged.
```

== Annotating the route

> Forwarded header handling is configured by annotating the `Route` object.

Reviewer: Probably makes sense to note that the annotations can also be set on `Ingress` objects which are managed by the ingresstoroute controller
